src/jsx.d.ts

```diff
@@ -1533,7 +1533,7 @@ export namespace JSXInternal {
 		| '_top'
 		| (string & {});
 
-	interface AnchorHTMLAttributes<T extends EventTarget>
+	interface AnchorHTMLAttributes<T extends EventTarget = HTMLAnchorElement>
```

Comment:
Can we just add TypeScript tests for `createElement` and `jsx` rendering so that we ensure we cover all bases here.

Reply:
Good call on the tests, forgot to push mine, but I'm not quite sure I follow re:JSX rendering in particular? JSX will always go through `IntrinsicElements` which always specifies the event target type:

```ts
interface IntrinsicElements { 
    button: ButtonHTMLAttributes<HTMLButtonElement>;
}
```

This PR wouldn't affect the behavior here at all, as there's no way you should be able to have a different element other than `HTMLButtonElement` when using `<button>`.

Let me know if I'm misunderstanding though, certainly possible

Reply:
Fair enough! Added a matching test case for JSX too then